Formula/pyenv.rb

```diff
@@ -47,6 +49,20 @@ def install
   end
 
   test do
-    shell_output("eval \"$(#{bin}/pyenv init -)\" && pyenv versions")
+    # Create a fake python version and executable.
+    shell_output("mkdir -p \"$(pyenv root)/versions/1.2.3/bin\"")
+    shell_output("echo 'echo hello' > \"$(pyenv root)/versions/1.2.3/bin/foo\"")
+    shell_output("chmod +x \"$(pyenv root)/versions/1.2.3/bin/foo\"")
```

Comment:
Let's try this:
```suggestion
    pyenv_root = shell_output("#{bin}/pyenv_root")
    pyenv_bin = Pathname("#{pyenv_root}/versions/1.2.3/bin")
    (pyenv_bin/"foo").write "echo hello"
    chmod "+x", pyenv_bin/"foo"
```
This way, we're not needlessly creating new processes just to start a shell to do things we could do in Ruby. Also might be a bit easier to read now that we don't have to keep escaping `"`s.